internal/keepers/filter.go

```diff
@@ -80,18 +81,27 @@ func (rc *reportCoordinator) Filter() func(types.UpkeepKey) bool {
 			// TODO: the key is constructed in the registry. splitting out the
 			// block number here is a hack solution that should be fixed asap.
 			// AUTO-1480
-			var blKey string
+			var blKey int
 			parts := strings.Split(string(key), "|")
-			if len(parts) == 2 {
-				blKey = parts[0]
+			if len(parts) == 2 && parts[0] != "" {
+				blKey, err = strconv.Atoi(parts[0])
+				if err != nil {
+					return false
+				}
 			}
 
-			// only apply filter if key block is after block in cache
-			if len(bl.TransmitBlockNumber) > 0 && blKey > string(bl.TransmitBlockNumber) {
-				return true
+			// Return false if empty
+			if bl.TransmitBlockNumber == "" {
+				return false
 			}
 
-			return false
+			transmitBlockNumber, err := strconv.Atoi(string(bl.TransmitBlockNumber))
+			if err != nil {
+				return false
+			}
+
+			// only apply filter if key block is after block in cache
```

Comment:
nit. comment is opposite. 
key if filtered if key block is less than cache transmit